### PR TITLE
added streaming responses. Fixed JSONP.

### PR DESCRIPTION
**Pull request by n0nag0n**

### CODE REVIEW

**Comment by fadrian06** [comment]
For future PRs you may give some description, isn't?...

**Comment by fadrian06** [comment]
image

File: .vscode/settings.json
Comment (fadrian06) [inline .vscode/settings.json:1]: Al fin !!!!!

Comment (fadrian06) [inline .vscode/settings.json:1]: Al fin !!!!!

File: flight/Engine.php [hunk 1]
```diff
@@ -364,15 +364,17 @@ public function path(string $dir): void
...
+     * @param Route $route The route to process the middleware for.
      * @param string $event_name If this is the before or after method.
      */
-    protected function processMiddleware(array $middleware, array $params, string $event_name): bool
+    protected function processMiddleware(Route $route, string $event_name): bool
     {
         $at_least_one_middleware_failed = false;
 
-        foreach ($middleware as $middleware) {
+        $middlewares = $event_name === 'before' ? $route->middleware : array_reverse($route->middleware);
```
Comment (fadrian06) [inline flight/Engine.php]: You could use Dispatcher::FILTER_BEFORE to avoid possible typos

File: flight/Engine.php [hunk 2]
```diff
@@ -481,18 +499,14 @@ public function _start(): void
                 $params
             );
 
-            if ($response->v2_output_buffering === false) {
+            if ($response->v2_output_buffering === false && $route->is_streamed === false) {
```
Comment (fadrian06) [inline flight/Engine.php]: You could extract that conditional in a separate method to avoid duplications

**Review by fadrian06 — APPROVED** [review]
Some docblocks are redundant, there was 2 tabs again, but it works :D

**Comment by n0nag0n** [comment]
Documentation here. https://github.com/flightphp/docs/pull/13

File: flight/Engine.php (hunk 1 above)
Comment (n0nag0n) [reply flight/Engine.php]: Deal

File: flight/Engine.php (hunk 2 above)
Comment (n0nag0n) [reply flight/Engine.php]: How about I'll pull it out to a variable. That'll be easiest since it's really only used in this one method

**Review by fadrian06 — APPROVED** [review]

**Comment by raindeer72** [comment]
Thank you

**Comment by raindeer72** [comment]
Thank you

**Comment by raindeer72** [comment]
Thank you

**Pull request merged**